src/Features/CSharp/EncapsulateField/CSharpEncapsulateFieldService.cs

```diff
@@ -130,6 +130,12 @@ internal class CSharpEncapsulateFieldService : AbstractEncapsulateFieldService
                                 .Where(f => f.Name.Length != 0);
         }
 
+        private bool CanEncapsulate(FieldDeclarationSyntax field)
+        {
+            return field.Parent is TypeDeclarationSyntax
+                || field.Parent is EnumDeclarationSyntax;
```

Comment:
You are right.  I was concerned with only fixing the problem that it not cause the code generator to assert/throw, so I basically copied the constraint from the code gen.  But really, this feature cannot work on enums either, so I'll remove that.